### PR TITLE
feat: UI improvements in Query Management UI

### PR DESCRIPTION
**Pull request by omkarK06**
#4936 

## Release Notes

- **New Features**
  - Introduced a new component, **RunningQueries**, for managing and displaying running queries with enhanced filtering options.
  - Added a **RunningQueriesList** component for displaying query data with pagination and action capabilities.

- **Improvements**
  - Enhanced layout and styling of the **QueryList** component for better user experience.
  - Streamlined data handling in **RunningQueriesList** by utilizing props and emitted events, improving flexibility.

- **Localization**
  - Added new localization keys for query metrics, including "Number of Queries," "Total Exec. Duration," "Total Query Range," and "Search Type," to improve user interface clarity.

### CODE REVIEW

File: web/src/components/queries/RunningQueriesList.vue [hunk 1]
```diff
@@ -364,28 +376,23 @@ export default defineComponent({
...
-          return queries.value.filter(
+          return props.rows.filter(
             (query: any) =>
               query.user_id.toLowerCase().includes(newVal.toLowerCase()) ||
               query.org_id.toLowerCase().includes(newVal.toLowerCase()) ||
               query.stream_type.toLowerCase().includes(newVal.toLowerCase()) ||
               query.status.toLowerCase().includes(newVal.toLowerCase()) ||
               query.trace_id.toLowerCase().includes(newVal.toLowerCase()) ||
-              query.work_group.toLowerCase().includes(newVal.toLowerCase())
+              query.work_group.toLowerCase().includes(newVal.toLowerCase()),
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/RunningQueriesList.vue:395]: Fixed

File: web/src/components/queries/RunningQueries.vue [hunk 2]
```diff
@@ -0,0 +1,827 @@
...
+      const averageQueryDuration = totalQueryDuration / queries.length;
+
+      return durationFormatter(averageQueryDuration); // You can also return the total if needed
+    };
+
+    const columns = ref<QTableProps["columns"]>([
+      {
+        name: "#",
+        label: "#",
+        field: "#",
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/RunningQueries.vue:421]: Fine for now

File: web/src/components/queries/SummaryList.vue [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+        });
+    };
+
+    const confirmDeleteAction = (props: any) => {
+      emit("delete:queries", props.row.trace_ids || []);
+    };
+
+    const handleMultiQueryCancel = () => {
+      emit("delete:queries");
+    };
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/SummaryList.vue:482]: As we have model for selected rows to delete, we don't need to explicitly send trace_ids

File: web/src/components/queries/SummaryList.vue [hunk 4]
```diff
@@ -0,0 +1,589 @@
...
+          });
+        })
+        .finally(() => {
+          dismiss();
+        });
+    };
+
+    const confirmDeleteAction = (props: any) => {
+      emit("delete:queries", props.row.trace_ids || []);
+    };
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/SummaryList.vue:478]: In summary list, it contains list of grouped queries in trace_ids

File: web/src/components/queries/RunningQueries.vue [hunk 5]
```diff
@@ -0,0 +1,827 @@
...
+
+<template>
+  <div class="running-queries-page q-pt-md q-px-md" v-if="isMetaOrg">
+    <div class="flex justify-between items-center full-width">
+      <div class="text-h6 q-my-xs" data-test="log-stream-title-text">
+        {{ t("queries.runningQueries") }}
+      </div>
+      <q-space />
+      <div class="flex items-start">
+        <div data-test="streams-search-stream-input" class="flex items-center">
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/RunningQueries.vue]: fixed 8cc0cfa0a85f816c17dc2440b1ee69501f6ad9f4

File: web/src/components/queries/RunningQueries.vue [hunk 6]
```diff
@@ -0,0 +1,827 @@
...
+      } else if (selectedSearchField.value === "query_range") {
+        return [
+          { label: "> 5 minutes", value: "gt_5m" },
+          { label: "> 10 minutes", value: "gt_10m" },
+          { label: "> 15 minutes", value: "gt_15m" },
+          { label: "> 1 hour", value: "gt_1h" },
+          { label: "> 1 day", value: "gt_1d" },
+          { label: "> 1 week", value: "gt_1w" },
+          { label: "> 1 Month", value: "gt_1M" },
+          { label: "> 1 Month", value: "gt_1M" },
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/RunningQueries.vue]: fixed 8cc0cfa0a85f816c17dc2440b1ee69501f6ad9f4

File: web/src/components/queries/SummaryList.vue [hunk 7]
```diff
@@ -0,0 +1,589 @@
...
+
+      rows.sort((a: any, b: any) => b.created_at - a.created_at);
+
+      return rows.map((row: any, index) => {
+        return {
+          "#": index < 9 ? `0${index + 1}` : index + 1,
+          user_id: row?.user_id,
+          org_id: row?.org_id,
+          duration: getDuration(row?.created_at),
+          queryRange: queryRange(row?.query?.start_time, row?.query?.end_time),
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/SummaryList.vue]: fixed 8cc0cfa0a85f816c17dc2440b1ee69501f6ad9f4

File: web/src/components/queries/SummaryList.vue [hunk 8]
```diff
@@ -0,0 +1,589 @@
...
+        return [
+          { label: "> 5 minutes", value: "gt_5m" },
+          { label: "> 10 minutes", value: "gt_10m" },
+          { label: "> 15 minutes", value: "gt_15m" },
+          { label: "> 1 hour", value: "gt_1h" },
+          { label: "> 1 day", value: "gt_1d" },
+          { label: "> 1 week", value: "gt_1w" },
+          { label: "> 1 Month", value: "gt_1M" },
+          { label: "> 1 Month", value: "gt_1M" },
+        ];
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/SummaryList.vue]: fixed 8cc0cfa0a85f816c17dc2440b1ee69501f6ad9f4

File: web/src/components/queries/SummaryList.vue [hunk 9]
```diff
@@ -0,0 +1,589 @@
...
+        sortable: true,
+        field: "queryRange",
+      },
+      {
+        name: "actions",
+        field: "actions",
+        label: t("common.actions"),
+        align: "center",
+      },
+    ]);
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/SummaryList.vue:224]: fixed 8cc0cfa0a85f816c17dc2440b1ee69501f6ad9f4

File: web/src/components/queries/RunningQueriesList.vue [hunk 10]
```diff
@@ -198,24 +131,106 @@
...
+        org_id: "otlp-production",
+        stream_type: "logs",
+        query: {
+          sql: 'SELECT histogram(_timestamp,\'43200 seconds\') AS "x_axis_1", COUNT("k8s_app_instance") AS "y_axis_1", "k8s_pod_name" AS "breakdown_1" FROM "default" GROUP BY "x_axis_1", "breakdown_1" ORDER BY "x_axis_1" ASC',
+          start_time: 1729771200000000,
+          end_time: 1729814400000000,
+        },
+        scan_stats: null,
+      },
+    ];
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/RunningQueriesList.vue]: fixed 8cc0cfa0a85f816c17dc2440b1ee69501f6ad9f4

File: web/src/components/queries/RunningQueriesList.vue [hunk 11]
```diff
@@ -198,24 +131,106 @@ import { durationFormatter } from "@/utils/zincutils";
...
+
+    const runningQueryTypes = [
+      { label: "User Summary", value: "summary" },
+      { label: "All Queries", value: "all" },
+    ];
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/RunningQueriesList.vue]: fixed 8cc0cfa0a85f816c17dc2440b1ee69501f6ad9f4

File: web/src/components/queries/QueryList.vue [hunk 12]
```diff
@@ -1,7 +1,7 @@
 <template>
-  <q-card class="column full-height no-wrap">
+  <q-card class="column full-height no-wrap" style="width: 40vw">
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/QueryList.vue]: fixed

File: web/src/components/queries/RunningQueriesList.vue [hunk 13]
```diff
@@ -20,10 +35,10 @@
         </div>
       </template>
       <template #header-selection="scope">
-        <q-checkbox v-model="scope.selected" size="sm" color="secondary" />
+        <q-checkbox v-model="scope.selected" size="xs" color="secondary" />
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/RunningQueriesList.vue:38]: This is fine, as we have reduced padding of rows in Query Management tables. So used xs size checkbox

File: web/src/components/queries/SummaryList.vue [hunk 14]
```diff
@@ -0,0 +1,302 @@
...
+  props: {
+    rows: {
+      type: Array,
+      required: true,
+    },
+    selectedRows: {
+      type: Array,
+      required: true,
+    },
+  },
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/SummaryList.vue:124]: Will introduce type for rows and selectedRows

File: web/src/components/queries/SummaryList.vue [hunk 15]
```diff
@@ -0,0 +1,274 @@
...
+  emits: [
+    "cancel:hideform",
+    "filter:queries",
+    "update:selectedRows",
+    "delete:queries",
+  ],
+  setup(props, { emit }) {
+    const { isMetaOrg } = useIsMetaOrg();
+    const resultTotal = ref<number>(0);
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/SummaryList.vue:122]: Done

**Review by neha00290 — APPROVED** [review]

File: web/src/components/queries/RunningQueries.vue [hunk 16]
```diff
@@ -0,0 +1,865 @@
...
+      summaryRows,
+    };
+  },
+});
+</script>
+
+<style lang="scss" scoped>
+.running-queries-page {
+  ::v-deep .q-table th,
+  ::v-deep .q-table td {
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/RunningQueries.vue:835]: Fixed

File: web/src/components/queries/RunningQueries.vue [hunk 17]
```diff
@@ -0,0 +1,865 @@
...
+          if (newVal.startsWith("lt_")) {
+            return timeDifference < timeMap[newVal];
+          } else if (newVal.startsWith("gt_")) {
+            return timeDifference > timeMap[newVal];
+          } else {
+            return true; // If the filter is not recognized, return all items
+          }
+        });
+      }
+    });
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/RunningQueries.vue]: Done!

File: web/src/components/queries/RunningQueries.vue [hunk 18]
```diff
@@ -0,0 +1,865 @@
...
+          ...user,
+          "#": index + 1,
+          duration: durationFormatter(user.duration),
+          queryRange: durationFormatter(user.queryRange),
+        }));
+      } catch (error) {
+        console.error("Error in getRunningQueriesSummary", error);
+        return [];
+      }
+    };
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/RunningQueries.vue:305]: Added Todo

File: web/src/components/queries/RunningQueriesList.vue [hunk 19]
```diff
@@ -1,15 +1,30 @@
...
       ref="qTable"
-      :rows="rowsQuery"
+      :rows="rows"
       :columns="columns"
       :pagination="pagination"
       row-key="trace_id"
       style="width: 100%"
       selection="multiple"
-      v-model:selected="selectedRow"
+      v-model:selected="selectedRowsModel"
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/RunningQueriesList.vue]: Done

File: web/src/components/queries/RunningQueriesList.vue [hunk 20]
```diff
@@ -364,237 +244,54 @@ export default defineComponent({
...
   },
 });
 </script>
 
-<style scoped>
+<style lang="scss" scoped>
+.query-management-tabs {
+  ::v-deep .q-btn:before {
+    border: none !important;
+  }
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/RunningQueriesList.vue]: Done

File: web/src/components/queries/SummaryList.vue [hunk 21]
```diff
@@ -0,0 +1,302 @@
...
+
+    const loadingState = ref(false);
+    const queries = ref([]);
+
+    const deleteDialog = ref({
+      show: false,
+      title: "Delete Running Query",
+      message: "Are you sure you want to delete this running query?",
+      data: null as any,
+    });
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/components/queries/SummaryList.vue]: Done

**Pull request merged**